File: src/main/java/club/deneb/client/features/modules/player/AutoWalk.java
package club.deneb.client.features.modules.player;

import club.deneb.client.features.Category;
import club.deneb.client.features.Module;
import club.deneb.client.features.ModuleManager;
import club.deneb.client.features.modules.render.Pathfind;
import club.deneb.client.utils.EntityUtil;
import club.deneb.client.value.Value;
import net.minecraft.pathfinding.PathPoint;
import net.minecraftforge.client.event.InputUpdateEvent;
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;


/**
 * Created by 086 on 16/12/2017.
 */
@Module.Info(name = "AutoWalk", category = Category.PLAYER)
public class AutoWalk extends Module {

    Value<String> mode = setting("Mode","Forward", listOf("Forward","Backwards","Path"));

    @SubscribeEvent
    public void onInput(InputUpdateEvent event){
        switch (mode.getValue()) {
            case "Forward":
                event.getMovementInput().moveForward = 1;
                break;
            case "Backwards":
                event.getMovementInput().moveForward = -1;
                break;
            case "Path":
                if (Pathfind.points.isEmpty()) return;
                event.getMovementInput().moveForward = 1;
                if (mc.player.isInWater() || mc.player.isInLava()) mc.player.movementInput.jump = true;
                else if (mc.player.collidedHorizontally && mc.player.onGround) mc.player.jump();
                if (!ModuleManager.getModuleByName("Pathfind").isEnabled() || Pathfind.points.isEmpty()) return;
                PathPoint next = Pathfind.points.get(0);
                lookAt(next);
                break;
        }
    }

    private void lookAt(PathPoint pathPoint) {
        double[] v = EntityUtil.calculateLookAt(pathPoint.x + .5f, pathPoint.y, pathPoint.z + .5f, mc.player);
        mc.player.rotationYaw = (float) v[0];
        mc.player.rotationPitch = (float) v[1];
    }

    private static enum AutoWalkMode {
        FORWARD, BACKWARDS, PATH
    }
}

File: src/main/java/club/deneb/client/features/modules/player/AutoWalk.kt
package club.deneb.client.features.modules.player

import club.deneb.client.features.Category
import club.deneb.client.features.Module
import club.deneb.client.features.ModuleManager
import club.deneb.client.features.modules.render.PathFind
import club.deneb.client.utils.EntityUtil
import club.deneb.client.value.ModeValue
import net.minecraft.pathfinding.PathPoint
import net.minecraftforge.client.event.InputUpdateEvent
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent

@Module.Info(name = "AutoWalk", category = Category.PLAYER)
class AutoWalk : Module() {

    private val mode: ModeValue<String> = setting("Mode", "Forward", listOf("Forward", "Backwards", "Path"))
    @SubscribeEvent
    fun onInput(event: InputUpdateEvent) {
        when (mode.value) {
            "Forward" -> event.movementInput.moveForward = 1f
            "Backwards" -> event.movementInput.moveForward = -1f
            "Path" -> {
                if (PathFind.points.isEmpty()) return
                event.movementInput.moveForward = 1f
                if (mc.player.isInWater || mc.player.isInLava) mc.player.movementInput.jump =
                    true else if (mc.player.collidedHorizontally && mc.player.onGround) mc.player.jump()
                if (!ModuleManager.getModuleByName("Pathfind").isEnabled || PathFind.points.isEmpty()) return
                val next = PathFind.points[0]
                lookAt(next)
            }
        }
    }

    private fun lookAt(pathPoint: PathPoint) {
        val v = EntityUtil.calculateLookAt(
            (pathPoint.x + .5f).toDouble(),
            pathPoint.y.toDouble(),
            (pathPoint.z + .5f).toDouble(),
            mc.player
        )
        mc.player.rotationYaw = v[0].toFloat()
        mc.player.rotationPitch = v[1].toFloat()
    }
}